CharlieBackend.Api/Controllers/SchedulesController.cs

```diff
@@ -37,74 +38,125 @@ public SchedulesController(IScheduleService scheduleService)
+        [Authorize(Roles = "Secretary, Admin, Mentor, Student")]
+        [HttpPost("events")]
+        public async Task<ActionResult<IList<ScheduledEventDTO>>> GetEventsFiltered(ScheduledEventFilterRequestDTO request)
+        {
+            var foundSchedules = await _scheduleService.GetEventsFiltered(request);
+
+            return foundSchedules.ToActionResult();
+        }
+
+        /// <summary>
+        /// Updates range of events
+        /// </summary>
+        /// <remarks>
+        /// Event date could not changed via this method.
+        /// Event start and Finish could only be used to change event start and finish time.
+        /// </remarks>
+        /// <response code="200">Successful update of schedule</response>
+        /// <response code="HTTP: 404, API: 3">Error, update data is missing</response>
+        /// <response code="HTTP: 400, API: 0">Error, update data is wrong</response>
+        [SwaggerResponse(200, type: typeof(EventOccurrence))]
+        [Authorize(Roles = "Secretary, Admin")]
+        [HttpPut("events/updateRange")]
+        public async Task<ActionResult<IList<ScheduledEventDTO>>> UpdateEventRange([FromBody]EventUpdateRangeDTO request)
+        {
+            var foundSchedules = await _scheduleService.UpdateEventsRange(request.Filter, request.Request);
+
+            return foundSchedules.ToActionResult();
+        }
+
+        /// <summary>
+        /// Updates a single event
         /// </summary>
-        /// <response code="200">Successful return of schedules list</response>
-        /// <response code="HTTP: 404, API: 3">Error, student group not found</response>
-        [SwaggerResponse(200, type: typeof(IList<ScheduleDto>))]
+        /// <response code="200">Successful update of schedule</response>
+        /// <response code="HTTP: 404, API: 3">Error, update data is missing</response>
+        /// <response code="HTTP: 400, API: 0">Error, update data is wrong</response>
+        [SwaggerResponse(200, type: typeof(EventOccurrenceDTO))]
         [Authorize(Roles = "Secretary, Admin")]
-        [HttpGet("{studentGroupId}/groupSchedule")]
-        public async Task<ActionResult<List<ScheduleDto>>> GetSchedulesByStudentGroupIdAsync(long studentGroupId)
+        [HttpPut("events/{scheduledEventID}")]
+        public async Task<ActionResult<ScheduledEventDTO>> UpdateEventById(long scheduledEventID, [FromBody]UpdateScheduledEventDto request)
         {
-            var foundSchedules = await _scheduleService.GetSchedulesByStudentGroupIdAsync(studentGroupId);
+            var foundSchedules = await _scheduleService.UpdateScheduledEventByID(scheduledEventID, request);
 
             return foundSchedules.ToActionResult();
         }
 
         /// <summary>
-        /// Updates shedule
+        /// Updates a single event
         /// </summary>
+        /// <remarks>
+        /// Old instance os event occurance is replaced with the new one (id is the same). 
```

Comment:
> Old instance **os** of event **occurance** is replaced with the new one (id is the same).

replace bold words:
os -> of 
occurance -> occurrence

Reply:
updated